### ISSUE TITLE
Dragging/maximizing the console wallet on windows causes it to crash

### ISSUE THREAD

**Issue by stringhandler** [tari-project/tari#6729]
To reproduce:

Open console wallet on windows.

Maximize it, then drag it to a new location.

Expected: maximizing and dragging works.

Actual: console wallet closes, I assume a panic.

NOTE: seems to happen on the second or third resize that happens

**Comment by hansieodendaal** [comment]
Performing a high level investigation showed that different results are obtained when using _Windows Console Host_ or _Windows Termimal_ as the default terminal application, or wether Windows 10 or Windows 11 are used.

**tldr;**   For the current **Console Wallet** application, running on **Windows 10** and using **Windows Console Host** is a bad combination.

![Image](https://github.com/user-attachments/assets/f97277f6-1943-4eda-ad70-bd2f08475cad)

- Using **Windows Console Host** as the defaiult terminal application: 
  - Windows 10 (22H2)
    - Console Wallet crashes on the first _maximize - drag_ or _maximize - reset_ attempt
  - Windows 11 (24H2)
    - Console Wallet never crashes on any first _maximize - drag_ or _maximize - reset_ attempt
 - Using **Windows Terminal** as the defaiult terminal application: 
    - Windows 10 (22H2)
      - Console Wallet never crashes on any first _maximize - drag_ or _maximize - reset_ attempt
    - Windows 11 (24H2)
      - Console Wallet never crashes on any first _maximize - drag_ or _maximize - reset_ attempt

**Comment by hansieodendaal** [comment]
Error details with Windows 10
```
2025-01-20 11:07:32.673156500 [wallet::app::crossterm_events] ERROR Internal error in crossterm events: No process is on the other end of the pipe. (os error 233)
2025-01-20 11:07:32.673416100 [wallet::app::crossterm_events] ERROR Internal error in crossterm events: No process is on the other end of the pipe. (os error 233)
2025-01-20 11:07:32.673449300 [wallet::app::crossterm_events] ERROR Internal error in crossterm events: No process is on the other end of the pipe. (os error 233)
2025-01-20 11:07:32.673469100 [wallet::ui::app] ERROR Error drawing interface. No process is on the other end of the pipe. (os error 233)
2025-01-20 11:07:32.673488200 [wallet::app::crossterm_events] ERROR Internal error in crossterm events: No process is on the other end of the pipe. (os error 233)

...

2025-01-20 11:07:32.923368000 [wallet::app::crossterm_events] ERROR Internal error in crossterm events: No process is on the other end of the pipe. (os error 233)
2025-01-20 11:07:32.923380000 [wallet::app::crossterm_events] ERROR Internal error in crossterm events: No process is on the other end of the pipe. (os error 233)
2025-01-20 11:07:32.923387800 [wallet::app::crossterm_events] INFO  Tick event channel shutting down
```

**Comment by hansieodendaal** [comment]
This test PR contains test code and some results: https://github.com/hansieodendaal/tari/pull/17

**tldr;** The backend crossterm process dies with maximize -> reset. This also holds for the latest release crossterm = { version = "0.28.1" }

**Comment by SWvheerden** [comment]
Going to close this for now, there are workarounds posted here above.
